RenderSystems/GLSupport/src/EGL/Wayland/OgreWaylandEGLSupport.cpp

```diff
@@ -0,0 +1,222 @@
+// This file is part of the OGRE project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at https://www.ogre3d.org/licensing.
+// SPDX-License-Identifier: MIT
+
+#include "OgreException.h"
+#include "OgreLogManager.h"
+#include "OgreRoot.h"
+#include "OgreStringConverter.h"
+
+#include "OgreWaylandEGLSupport.h"
+#include "OgreWaylandEGLWindow.h"
+
+#include "OgreGLUtil.h"
+
+namespace Ogre
+{
+
+GLNativeSupport* getGLSupport(int profile) { return new WaylandEGLSupport(profile); }
+
+const wl_registry_listener WaylandEGLSupport::mRegistryListener = {
+    .global = WaylandEGLSupport::globalRegistryHandler, .global_remove = WaylandEGLSupport::globalRegistryRemover};
+
+const wl_output_listener WaylandEGLSupport::mOutputListener = {
+    .geometry = WaylandEGLSupport::output_handle_geometry,
+    .mode = WaylandEGLSupport::output_handle_mode,
+  };
+
+void WaylandEGLSupport::globalRegistryHandler(void* data, struct wl_registry* registry, uint32_t id,
+                                              const char* interface, uint32_t version)
+{
+    LogManager::getSingleton().logMessage("Got a registry event for " + std::string(interface) + " " +
+                                          std::to_string(id)); //, LogMessageLevel::LML_TRIVIAL);
+
+    auto ptr = static_cast<WaylandEGLSupport*>(data);
+
+    if (std::string(interface) == std::string(wl_compositor_interface.name))
+    {
+        ptr->mWlCompositor = static_cast<wl_compositor*>(wl_registry_bind(registry, id, &wl_compositor_interface, 4));
+    } else if (std::string(interface) == std::string(wl_output_interface.name)) {
+      ptr->mWlOutput = static_cast<wl_output*>(wl_registry_bind(registry, id, &wl_output_interface, 1));
+      wl_output_add_listener(ptr->mWlOutput, &WaylandEGLSupport::mOutputListener, ptr);
+    }
+}
+
+void WaylandEGLSupport::globalRegistryRemover(void* data, struct wl_registry* registry, uint32_t id)
+{
+    // auto ptr = static_cast<WaylandEGLSupport*>(data);
+    LogManager::getSingleton().logMessage("Got a registry losing event for " + std::to_string(id));
+}
+
+void WaylandEGLSupport::output_handle_geometry(
+    void *data, wl_output *wl_output,
+    int32_t x, int32_t y,
+    int32_t physical_width, int32_t physical_height,
+    int32_t subpixel,
+    const char *make, const char *model,
+    int32_t output_transform)
+{}
+
+void WaylandEGLSupport::output_handle_mode(
```

Comment:
I believe it's best to avoid implementing this feature within Ogre. As with fullscreen I would rather direct users to SDL_GetDisplayMode

Reply:
Removed.

Reply:
fore reference, this is the WIP code I had for this: https://github.com/OGRECave/ogre/pull/3144

Reply:
I tested your WIP code and it works fine with a wayland window.